src/engine/filters/index.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  AudienceDefinitionFilter,
+  EngineConditionQuery,
+  PageFeatureResult,
+} from '../../../types';
+import {
+  versionMatches,
+  arrayIntersectsCondition,
+  vectorDistanceCondition,
+  cosineSimilarityCondition,
+} from './conditions';
+
+export const matchesQuery = (
+  query: EngineConditionQuery<AudienceDefinitionFilter>,
+  queryFeatures: PageFeatureResult
```

Comment:
I think the naming here can be changed, `query` and `queryFeatures` sounds like they come from the same object i.e `queryFeature` is a subset of `query`.

PageViewFeatures or PageFeatures I feel is a better way to describe what is stored client side based on user page visits/

Reply:
I agree